test/quantum_objects.jl

```diff
@@ -1,12 +1,50 @@
 @testset "Quantum Objects" begin
-    # unsupported size of array
-    for a in [rand(ComplexF64, 3, 2), rand(ComplexF64, 2, 2, 2)]
-        for t in [nothing, Ket, Bra, Operator, SuperOperator, OperatorBra, OperatorKet]
-            @test_throws DomainError Qobj(a, type = t)
+    # Qobj type inference
+    for T in [ComplexF32, ComplexF64]
```

Comment:
From now on I would put all the type-inference tests inside an inner `testset`

Reply:
okay, I will update it later